indexer/services/ender/src/handlers/deleveraging-handler.ts

```diff
@@ -1,18 +1,160 @@
+      `${this.eventType}_${deleveragedSubaccountUuid}_${perpetualMarket.clobPairId}`,
+      // To ensure that SubaccountUpdateEvents and OrderFillEvents for the same subaccount are not
+      // processed in parallel
+      `${SUBACCOUNT_ORDER_FILL_EVENT_TYPE}_${offsettingSubaccountUuid}`,
+      `${SUBACCOUNT_ORDER_FILL_EVENT_TYPE}_${deleveragedSubaccountUuid}`,
+      // To ensure that StatefulOrderEvents and OrderFillEvents for the same order are not
+      // processed in parallel
+      `${DELEVERAGING_EVENT_TYPE}_${offsettingSubaccountUuid}`,
+      `${DELEVERAGING_EVENT_TYPE}_${deleveragedSubaccountUuid}`,
+    ];
+  }
+
+  protected generateConsolidatedKafkaEvent(
+    subaccountIdProto: IndexerSubaccountId,
+    position: UpdatedPerpetualPositionSubaccountKafkaObject | undefined,
+    fill: FillFromDatabase,
+    perpetualMarket: PerpetualMarketFromDatabase,
+  ): ConsolidatedKafkaEvent {
+    const message: SubaccountMessageContents = {
+      fills: [
+        generateFillSubaccountMessage(fill, perpetualMarket.ticker),
+      ],
+      perpetualPositions: position === undefined ? undefined : generatePerpetualPositionsContents(
+        subaccountIdProto,
+        [position],
+        perpetualMarketRefresher.getPerpetualMarketsMap(),
+      ),
+    };
+    return this.generateConsolidatedSubaccountKafkaEvent(
+      JSON.stringify(message),
+      subaccountIdProto,
+      undefined,
+      true,
+      message,
+    );
+  }
+
+  protected generateTradeKafkaEventFromDeleveraging(
+    fill: FillFromDatabase,
+  ): ConsolidatedKafkaEvent {
+    const tradeContents: TradeMessageContents = {
+      trades: [
+        {
+          id: fill.eventId.toString('hex'),
+          size: fill.size,
+          price: fill.price,
+          side: fill.side.toString(),
+          createdAt: fill.createdAt,
+          liquidation: false,
+          deleveraging: true,
+        },
+      ],
+    };
+    return this.generateConsolidatedTradeKafkaEvent(
+      JSON.stringify(tradeContents),
+      fill.clobPairId,
+    );
   }
 
   // eslint-disable-next-line @typescript-eslint/require-await
   public async internalHandle(): Promise<ConsolidatedKafkaEvent[]> {
-    // Implement this
-    return [];
+    const eventDataBinary: Uint8Array = this.indexerTendermintEvent.dataBytes;
+    const transactionIndex: number = indexerTendermintEventToTransactionIndex(
+      this.indexerTendermintEvent,
+    );
+    const result: pg.QueryResult = await storeHelpers.rawQuery(
+      `SELECT dydx_deleveraging_handler(
+        ${this.block.height}, 
+        '${this.block.time?.toISOString()}', 
+        '${JSON.stringify(DeleveragingEventV1.decode(eventDataBinary))}', 
+        ${this.indexerTendermintEvent.eventIndex}, 
+        ${transactionIndex}, 
+        '${this.block.txHashes[transactionIndex]}' 
+      ) AS result;`,
+      { txId: this.txId },
+    ).catch((error: Error) => {
+      logger.error({
+        at: 'deleveragingHandler#handleViaSqlFunction',
```

Comment:
Why `deleveragingHandler` and not `DeleveragingHandler`?

I saw this in other files. Is this common to lower case the first character when logging?

Reply:
I was just following the convention in other files. But DeleveragingHandler makes more sense to me.